### PR TITLE
fix: scope codegraph batch complexity targets to file paths

### LINKED ISSUE #1721: codegraph batch complexity ignores file-path targets, returns global summary for every file
## Summary

`codegraph batch complexity <file1> <file2> ...` (documented in CLAUDE.md as the way to batch-query multi-target complexity, and used by the `/titan-gauntlet` skill for per-batch metrics) returns identical, unfiltered whole-repo results for every target instead of per-file data.

## Repro

```
$ codegraph batch complexity src/extractors/javascript.ts src/extractors/groovy.ts -T --json
```

Output (trimmed):
```json
{
  "results": [
    { "target": "src/extractors/javascript.ts", "data": { "functions": [], "summary": { "analyzed": 3447, "maxCognitive": 110, ... } } },
    { "target": "src/extractors/groovy.ts",     "data": { "functions": [], "summary": { "analyzed": 3447, "maxCognitive": 110, ... } } }
  ]
}
```

Both targets return `functions: []` and an *identical* `summary` (repo-wide `analyzed: 3447`, `maxCognitive: 110`) — the same numbers regardless of which file is queried.

Compare to the direct single-file command, which works correctly:
```
$ codegraph complexity --file src/extractors/javascript.ts --health -T --json
```
This correctly returns `extractDestructuredBindingsWalk` (cognitive 110), `handleVariableDecl` (cognitive 107), etc., scoped to that file only.

## Root cause

`src/features/batch.ts` registers `complexity` with `sig: 'dbOnly'`:
```ts
complexity: { fn: complexityData as (...args: unknown[]) => unknown, sig: 'dbOnly' },
```
and the `dbOnly` dispatch branch in `batchData()` calls:
```ts
data = entry.fn(customDbPath, { ...opts, target });
```
passing the CLI-supplied file path as `opts.target`.

But `complexityData` (`src/features/complexity-query.ts:206-239`) treats `opts.target` as a **symbol/function-name** filter and `opts.file` as the **file-path** filter:
```ts
const { where, params } = buildComplexityWhere({
  noTests,
  target: opts.target || null,      // <- file path lands here
  fileFilter: opts.file || null,    // <- never populated by batch.ts
  kindFilter: opts.kind || null,
});
```

Since the file path never matches any symbol name, the `target`-scoped `functions` list comes back empty, and the `summary` aggregate appears to ignore the (failed) filter entirely and fall back to whole-repo stats — hence identical numbers for every target.

## Impact

Any consumer of `codegraph batch complexity <files...>` — including the `/titan-gauntlet` skill's batch-metrics step and any other automation following the documented `codegraph batch t1 t2 t3 --json` pattern — silently gets wrong data (empty function lists + misleading global summary) instead of an error, which is worse than a crash because it looks superficially valid.

## Suggested fix

In `src/features/batch.ts`, when `command === 'complexity'` (or more generally for any `dbOnly` command whose targets are file paths rather than symbol names), pass the target through `opts.file` instead of `opts.target` — or add a way for each `BATCH_COMMANDS` entry to specify which opts key its targets map to.

## Workaround used

Per-file `codegraph complexity --file <f> --health -T --json` calls (one per file) instead of `codegraph batch complexity`.

### PR DESCRIPTION
**Pull request by carlos-alm**
## Summary
- `codegraph batch complexity <files...>` passed each file path through `opts.target` (a symbol-name filter), never `opts.file` (the file-path filter), so every target silently fell back to whole-repo stats instead of erroring.
- Added a general `targetKey` field on `BatchCommandEntry` so `dbOnly` commands can declare which `opts` key their targets map to; `complexity` now declares `targetKey: 'file'`.
- Added a regression test in `tests/integration/batch.test.ts` asserting distinct file targets return distinct, correctly-scoped results.

Closes #1721

## Test plan
- `npm test` — full suite green (3349 passed, 0 failed)
- `npm run lint` — clean
- Manual: `node dist/cli.js batch complexity <fileA> <fileB> -T --json` now matches per-file `codegraph complexity --file <f>` output

### CODE REVIEW

File: tests/integration/batch.test.ts [hunk 1]
```diff
@@ -197,12 +210,46 @@ describe('batchData — edge cases', () => {
 // ─── complexity (dbOnly sig) ─────────────────────────────────────────
 
 describe('batchData — complexity (dbOnly signature)', () => {
-  test('complexity command uses target as opts.target', () => {
-    const data = batchData('complexity', ['authenticate'], dbPath);
-    expect(data.total).toBe(1);
-    // complexityData returns functions array — it won't error for unknown targets
+  test('complexity command scopes each target to a file, not a symbol name (#1721)', () => {
```
[the remark this answers is not in the document]
Comment (carlos-alm) [reply tests/integration/batch.test.ts:213]: Checked this — there's no actual mismatch. The PR body says "Closes #1721" (not #1806), and the test at lines 213-214 consistently cites #1721 in both the test name and the inline comment. #1721 ("codegraph batch complexity ignores file-path targets, returns global summary for every file") is the issue this PR fixes. #1806 is an unrelated follow-up about `checkNoSignatureChanges` filed from a different PR's review. No code change needed here.

**Comment by carlos-alm** [comment]
@greptileai
